### PR TITLE
Port release automation fixes from liquibase/liquibase-percona#110

### PR DESCRIPTION
**Pull request by adangel**
@mcred 
These are the changes I did for liquibase-percona which probably also need to be applied here for postgresql.

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-1743) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by r2liquibase** [comment]
Hey @nvoxland , let me know your thoughts on this. If it's accepted I can work on rolling it out to the other extensions. 

Thanks, @adangel !

**Comment by mcred** [comment]
For all of the changes to maven commands with the additional `-B` option, the commands are already non-interactive. What additional benefit is there to running them in batch-mode?

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
         java-version: ${{ matrix.java }}
         distribution: 'adopt'
     - name: Run Unit Tests With Maven
-      run: mvn surefire:test --file pom.xml
```
Comment (mcred) [inline .github/workflows/pull-request.yml:35]: This is intentionally only running the unit tests and not the integration tests. Only release candidate PRs run integration tests. All PRs have the unit tests ran.

**Comment by adangel** [comment]
> For all of the changes to maven commands with the additional `-B` option, the commands are already non-interactive. What additional benefit is there to running them in batch-mode?

That's right, the used plugins are already non-interactive. This option also disabled transfer progress while downloading maven dependencies, which makes the log file smaller and easier to read. E.g. https://github.com/liquibase/liquibase-postgresql/runs/2576578700?check_suite_focus=true#step:5:8

In the below snippet, you won't get the lines, starting with "Progress(1)":

```
Downloading from central: https://repo.maven.apache.org/maven2/org/sonatype/plugins/nexus-staging-maven-plugin/1.6.8/nexus-staging-maven-plugin-1.6.8.pom
Progress (1): 2.7/12 kB
Progress (1): 5.5/12 kB
Progress (1): 8.2/12 kB
Progress (1): 11/12 kB 
Progress (1): 12 kB   
                   
Downloaded from central: https://repo.maven.apache.org/maven2/org/sonatype/plugins/nexus-staging-maven-plugin/1.6.8/nexus-staging-maven-plugin-1.6.8.pom (12 kB at 9.9 kB/s)
```

Only "Downloading ..." and "Downloaded ..." is left.

You could achieve the same with `--no-transfer-progress`, but batch mode is usually used when running maven in a CI system.

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (adangel) [reply .github/workflows/pull-request.yml:35]: The problem with `mvn surefire:test` is, that compile and test-compile is skipped, which means: there are no tests to run.
See https://github.com/liquibase/liquibase-postgresql/runs/2576578700?check_suite_focus=true#step:5:4310

```
[INFO] --- maven-surefire-plugin:2.12.4:test (default-cli) @ liquibase-postgresql ---
... lots of download message ...
[INFO] No tests to run.
```

Wwith `mvn clean test`, the [maven lifecycle](http://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#Lifecycle_Reference) is executed till the test phase, which runs the unit tests after the code has been compiled. But it doesn't execute the integration tests, which come in a later phase.

Also, running a plugin from command line directly like `mvn surefire:test` uses a different execution configuration (namely `default-cli`) - this could also be a reason, why no tests have been found to run (I don't know, what the defaults there would be for includes...)

So in summary: `mvn clean test` runs the unit tests, `mvn clean verify` runs unit tests+integration tests.
Note I used "verify" here, because sometimes the results of the integration tests are only evaluated in verify (e.g. the failsafe plugin does it that way).

**Review by mcred — APPROVED** [review]
